test/Microsoft.AspNet.Mvc.IntegrationTests/ValidationIntegrationTests.cs

```diff
@@ -464,6 +462,549 @@ private class Order5
+
+            // Assert
+            Assert.NotNull(modelBindingResult);
+            Assert.True(modelBindingResult.IsModelSet);
+
+            var model = Assert.IsType<Order7>(modelBindingResult.Model);
+            Assert.Equal("billybob", model.Customer.Name);
+
+            Assert.Equal(1, modelState.Count); // This fails due to #2446
+            Assert.Equal(1, modelState.ErrorCount);
+            Assert.False(modelState.IsValid);
+
+            var entry = Assert.Single(modelState, e => e.Key == "parameter.Customer.Name").Value;
+            Assert.Equal("billybob", entry.Value.AttemptedValue);
+            Assert.Equal("billybob", entry.Value.RawValue);
+
+            var error = Assert.Single(entry.Errors);
+            Assert.Equal("Too Long.", error.ErrorMessage);
+            Assert.Null(error.Exception);
+        }
+
+        [Fact(Skip = "Extra ModelState key because of #2446")]
+        public async Task Validation_StringLengthAttribute_OnPropertyOfNestedPOCO_NoData()
+        {
+            // Arrange
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "parameter",
+                ParameterType = typeof(Order7)
+            };
+
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext(request =>
+            {
+                request.QueryString = new QueryString("?");
+            });
+
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            var modelBindingResult = await argumentBinder.BindModelAsync(parameter, modelState, operationContext);
+
+            // Assert
+            Assert.NotNull(modelBindingResult);
+            Assert.True(modelBindingResult.IsModelSet);
+
+            var model = Assert.IsType<Order7>(modelBindingResult.Model);
+            Assert.Null(model.Customer);
+
+            Assert.Equal(0, modelState.Count); // This fails due to #2446
+            Assert.Equal(0, modelState.ErrorCount);
+            Assert.True(modelState.IsValid);
+        }
+
+        private class Order8
+        {
+            [ValidatePerson8]
+            public Person8 Customer { get; set; }
+        }
+
+        private class Person8
+        {
+            public string Name { get; set; }
+        }
+
+        private class ValidatePerson8Attribute : ValidationAttribute
+        {
+            protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+            {
+                if (((Person8)value).Name == "bill")
+                {
+                    return null;
+                }
+                else
+                {
+                    return new ValidationResult("Invalid Person.");
+                }
+            }
+        }
+
+        [Fact(Skip = "Extra ModelState key because of #2446")]
+        public async Task Validation_StringLengthAttribute_OnPOCOProperty_Valid()
```

Comment:
`Validation_CustomAttribute_OnPOCOProperty_Valid` ? There is no `StringLengthAttribute` here.